cmd/cc-uploader/main.go

```diff
@@ -189,16 +131,87 @@ func initializeServer(logger lager.Logger, uploaderConfig config.UploaderConfig,
 			tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
 		}
 
-		server = &http.Server{
-			Addr:      uploaderConfig.MutualTLS.ListenAddress,
-			Handler:   ccUploaderHandler,
-			TLSConfig: clientTLSConfig,
-		}
-		return server, http_server.NewTLSServer(uploaderConfig.MutualTLS.ListenAddress, ccUploaderHandler, clientTLSConfig)
+		return http_server.NewTLSServer(uploaderConfig.MutualTLS.ListenAddress, ccUploaderHandler, clientTLSConfig)
 	}
-	server = &http.Server{
-		Addr:    uploaderConfig.ListenAddress,
-		Handler: ccUploaderHandler,
+
+	return http_server.New(uploaderConfig.ListenAddress, ccUploaderHandler)
+}
+
+func waitForDrainingToFinish() <-chan struct{} {
+	done := make(chan struct{})
+	go func() {
+		uploadWaitGroup.Wait() // wait for all in-flight uploads to call Done()
+		close(done)            // signal completion
+	}()
+	return done
+}
+
+func configureServers(logger lager.Logger, uploaderConfig config.UploaderConfig, reconfigurableSink *lager.ReconfigurableSink) ifrit.Process {
+
+	var nonTLSRunner ifrit.Runner
+	tlsRunner := initializeServer(logger, uploaderConfig, true)
+	members := grouper.Members{
+		{Name: "cc-uploader-tls", Runner: tlsRunner},
+	}
+	if !uploaderConfig.DisableNonTLS {
+		nonTLSRunner = initializeServer(logger, uploaderConfig, false)
+		members = append(grouper.Members{
```

Comment:
is the "prepend" intened? If not, you can just write `members = append(members, grouper.Member{...})`, which might be more readable.

Reply:
It was just another way (ifrit way) to append the members in the correct order to start and top. Can also be done like you suggest by taking care of the correct append order.